core/src/main/java/org/ehcache/core/internal/service/ServiceLocator.java

```diff
@@ -445,11 +477,11 @@ ServiceMap lookupDependenciesOf(ServiceMap resolved, Class<? extends Service> re
 
     for (Class<?> interfaceClazz : clazz.getInterfaces()) {
       if (Service.class.isAssignableFrom(interfaceClazz)) {
-        dependencies.addAll(identifyImmediateDependenciesOf(Service.class.getClass().cast(interfaceClazz)));
+        dependencies.addAll(identifyImmediateOptionalDependenciesOf(Service.class.getClass().cast(interfaceClazz)));
```

Comment:
It feels like we could recurse over `lookupDependenciesOf` to prevent the code duplication between optional and not optional. Or keep the code mechanic and pass a lambda to have an optional/non-optional behavior.